src/queries/searchResultQuery.ts

```diff
@@ -0,0 +1,31 @@
+class SearchResultQuery {
+  queryParams: any;
+  searchResultQuery: any;
+
+  constructor(queryParams: any = {}) {
+    this.queryParams = queryParams;
+    this.searchResultQuery = {};
+  }
+
+  construct() {
+    this.filterByUserId();
+    this.filterByKeywords();
+
+    return this.searchResultQuery;
+  }
+
+  filterByUserId() {
```

Comment:
How about making these classes private:

```suggestion
  #filterByUserId() {
```

Or just using the standard prefix `_` to explicitly tell that these methods should not be accessed:

```suggestion
  _filterByUserId() {
```